src/Processors/QueryPlan/Optimizations/optimizeUseNormalProjection.cpp

```diff
@@ -187,7 +188,11 @@ std::optional<String> optimizeUseNormalProjections(Stack & stack, QueryPlan::Nod
         if (!analyzed)
             continue;
 
-        if (candidate.sum_marks >= ordinary_reading_marks)
+        if (candidate.sum_marks > ordinary_reading_marks)
+            continue;
+
+        /// When force_optimize_projection are true, consider projection when equal cost.
```

Comment:
To be honest, I don't think this fully addresses #71843. As I understand it, the issue is about *always* choosing a projection if it's logically valid, regardless of whether it's more expensive. That's different from `force_primary_key`, which is safe because the primary key is always at least as efficient, if not better. In contrast, forcing projection usage may result in worse performance.

A more robust and backward-compatible solution would be to introduce a new setting, e.g., `force_optimize_projection = 2`, which explicitly forces the use of *some* projection whenever possible, even if it's not cost-effective.